masonry/src/properties/gap.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::any::TypeId;
+
+use crate::core::{Property, UpdateCtx};
+use crate::properties::types::Length;
+
+/// The distance between two adjacent widgets in a [`Flex`](crate::widgets::Flex) or
+/// a [`Grid`](crate::widgets::Grid).
+///
+/// Equivalent to the css [gap] property.
+///
+/// ## Note on spacers and `Flex` widgets
+///
+/// This gap is between any two children, including `Flex` spacers.
+/// As such, using a non-zero gap and also adding may lead to counter-intuitive results.
+/// You should usually pick one or the other.
+///
+/// [gap]: https://developer.mozilla.org/en-US/docs/Web/CSS/gap
+#[expect(missing_docs, reason = "field names are self-descriptive")]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Gap {
+    pub gap: Length,
```

Comment:
Self-descriptive or not, I think it's less noisy to actually have a short doc comment.

```suggestion
#[derive(Clone, Copy, Debug, PartialEq)]
pub struct Gap {
    /// The distance between two adjacent widgets.
    pub gap: Length,
```